### PR TITLE
[22.05] Fix typos in test linter

### PR DESCRIPTION
**Pull request by bernt-matthias**
## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/tool_util/linters/tests.py [hunk 1]
```diff
@@ -97,9 +97,9 @@ def lint_tsts(tool_xml, lint_ctx):
             discover_datasets = corresponding_output.find(".//discover_datasets")
             if discover_datasets is not None:
                 if output.tag == "output":
-                    if "count" not in output.attrib and output.find("./discovered_datasets") is None:
+                    if "count" not in output.attrib and output.find("./discovered_dataset") is None:
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/linters/tests.py:100]: I quick grep of the codebase seems to show that "discover_datasets" is the right version.

**Review by nsoranzo — COMMENTED** [review]
I see there is another typo in `lib/galaxy/tool_util/xsd/galaxy.xsd`: `discover_data_sets`

File: lib/galaxy/tool_util/linters/tests.py (hunk 1 above)
Comment (bernt-matthias) [reply lib/galaxy/tool_util/linters/tests.py:100]: No, `discover_datasets` is child of the `data` tag in `outputs`:

[tool > outputs > data > discover_datasets](https://docs.galaxyproject.org/en/master/dev/schema.html#tool-outputs-data-discover-datasets)

and `discovered_dataset` is the child of `output` tag is `tests`

[tool > tests > test > output > discovered_dataset](https://docs.galaxyproject.org/en/master/dev/schema.html#tool-tests-test-output-discovered-dataset)

**Review by jmchilton — APPROVED** [review]

**Review by nsoranzo — APPROVED** [review]
I've added a commit fixing a few more ``discovered_datasets`` typos, if you want to double check them.

**Comment by bernt-matthias** [comment]
> I've added a commit fixing a few more `discovered_datasets` typos, if you want to double check them.

Looks good to me. Thanks for double checking.

**Pull request merged**